pkg/services/publicdashboards/service/service.go

```diff
@@ -342,6 +342,32 @@ func (pd *PublicDashboardServiceImpl) Delete(ctx context.Context, orgId int64, u
 	return nil
 }
 
+func (pd *PublicDashboardServiceImpl) HandleDashboardDeleted(ctx context.Context, dashboard *dashboards.Dashboard) error {
+	if dashboard.IsFolder {
+		// get all pubdashes for the folder
+		pubdashes, err := pd.store.GetByDashboardFolder(ctx, dashboard)
+		if err != nil {
+			return err
+		}
+		// delete each pubdash
+		for _, pubdash := range pubdashes {
+			err = pd.serviceWrapper.HandlePublicDashboardDeleted(ctx, pubdash)
+			if err != nil {
+				return err
+			}
+		}
+
+		return nil
+	}
+
+	pubdash, err := pd.store.FindByDashboardUid(ctx, dashboard.OrgID, dashboard.UID)
+	if err != nil {
+		return err
+	}
+
+	return pd.serviceWrapper.HandlePublicDashboardDeleted(ctx, pubdash)
```

Comment:
this is the only code that should be in this method

Reply:
The only implementations that differ are `HandlePublicDashboardDeleted`. All the code above this is okay to stay since it doesn't change between OSS and enterprise.